### PR TITLE
- PH-4624 Add AMQP API for message sending in Connector

### PR DESCRIPTION
**Pull request by pmalon**
- Improved Rabbit listener retry mechanism - now there is an easy way to define non retryable exception
- Added support for javax validation on Rabbit listeners
- Removed TransactionManager from Rabbit listener configuration - now each listener should be annotated with @Transactional to enable transactional message processing
- Migrated ATs to ITs
- Added javax validation annotations to the Event class

### CODE REVIEW

File: configuration/src/main/java/com/phorest/events/configuration/retry/DelayedRetryMessageRecoverer.java [hunk 1]
```diff
@@ -34,14 +50,27 @@ public void recover(Message message, Throwable cause) {
         String originalRoutingKey = message.getMessageProperties().getReceivedRoutingKey();
         int retryCount = (int) headers.getOrDefault(X_RETRY_COUNT, 0);
 
-        if (retryCount < maxRetryCount) {
+        if (shouldRetry(retryCount, cause)) {
             log.warn("Retrying to deliver for routing key {}. Attempts: {}, reason: {}", originalRoutingKey, retryCount, cause.getMessage());
```
Comment (kdomagal) [inline configuration/src/main/java/com/phorest/events/configuration/retry/DelayedRetryMessageRecoverer.java]: as far as I remember that cause.getMessage only says "due to listener thrown exception" something like that - no details on the exception though unless you show the stack trace :/

Comment (pmalon) [inline configuration/src/main/java/com/phorest/events/configuration/retry/DelayedRetryMessageRecoverer.java]: yup, but we log full exception after last retry, so I think that it's fine

Comment (pauldailly) [inline configuration/src/main/java/com/phorest/events/configuration/retry/DelayedRetryMessageRecoverer.java]: Can we do something like `cause?.cause?.message` so log contains more useful message?

Comment (pmalon) [inline configuration/src/main/java/com/phorest/events/configuration/retry/DelayedRetryMessageRecoverer.java]: I can try to do sth like it but in java way...

**Pull request merged**